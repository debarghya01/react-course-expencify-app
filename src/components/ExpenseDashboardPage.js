import React from 'react';
import ReactDOM from 'react-dom';
import ExpensList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';


const ExpenseDashboardPage = () => {
    return(
        <div>
            <ExpenseListFilters />
            <ExpensList />
        </div>
    )
}

export default ExpenseDashboardPage;