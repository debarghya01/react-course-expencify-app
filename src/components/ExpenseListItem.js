import React from 'react';
import { Link } from 'react-router-dom';


const ExpenseListItem = (props) => {
    
    return(
        <div>
            <Link to={`/edit/${props.id}`} ><p>  {`Expense ${props.description}`}</p> </Link> <p>{`${props.amount} ${props.createdAt}`} </p> 

        </div>
    )
}

export default ExpenseListItem;
