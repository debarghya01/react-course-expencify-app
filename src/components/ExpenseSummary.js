import React from 'react';
import {connect} from 'react-redux';
import numeral from 'numeral';
import getVisibleExpenses from './../selectors/expenses';
import getExpenseTotal from './../selectors/expenses-total';

export const ExpenseSummary = (props) => {
    return(
        <div>
            <h2>
                {`Viewing ${props.count} expenses totaling ${props.amount}`} 
            </h2>
        </div>
    )
} 

const mapStateToProps = (state) => {
    const expenses = getVisibleExpenses(state.expenses, state.filters)
    let total = getExpenseTotal(expenses)
    total = total/100;
    const formatedTotal = numeral(total).format('$0,0.00');
    return(
        {
            count: expenses.length,
            amount: formatedTotal
        }
    )
}

const ConnectedExpenseSummary = connect(mapStateToProps)(ExpenseSummary);
export default ConnectedExpenseSummary;