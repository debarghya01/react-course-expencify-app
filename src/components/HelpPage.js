import React from 'react';
import RecatDOM from 'react-dom';

const HelpPage = () => {
    return(
        <div>Help page</div>
    )
}

export default HelpPage;
