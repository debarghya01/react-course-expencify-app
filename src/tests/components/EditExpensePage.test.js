import React from 'react';
import {shallow} from 'enzyme';
import {EditExpensePage} from '../../components/EditExpensePage';
import expenses from '../fixtures/expenses';

let editExpense, removeExpense, history, wrapper;

beforeEach(() => {
    editExpense = jest.fn()
    removeExpense = jest.fn()
    history = {
        push: jest.fn()
    }
    wrapper = shallow(<EditExpensePage 
        expense={expenses[0]}
        editExpense={editExpense} 
        removeExpense={removeExpense} 
        history={history} />)
})

test('testing Edit Expense Page rendering', () => {
    expect(wrapper).toMatchSnapshot()
 })

test('testing Edit Expense Page rendering with editExpense spies', () => {
    wrapper.find('ExpenseForm').prop('onSubmit')(expenses[0])
    expect(editExpense).toHaveBeenLastCalledWith(expenses[0].id, expenses[0])
    expect(history.push).toHaveBeenLastCalledWith('/')
})

test('testing Edit Expense Page rendering with removeExpense spies', () => {
    wrapper.find('button').simulate('click')
    expect(removeExpense).toHaveBeenLastCalledWith({id: expenses[0].id})
    expect(history.push).toHaveBeenLastCalledWith('/')
})