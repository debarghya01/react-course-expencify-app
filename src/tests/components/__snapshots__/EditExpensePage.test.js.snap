// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Edit Expense Page rendering 1`] = `
<div>
  <ExpenseForm
    expense={
      Object {
        "amount": 2300,
        "createdAt": 0,
        "description": "Gum",
        "id": "1",
        "note": "",
      }
    }
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
     Remove Expense 
  </button>
</div>
`;
